File: backend/app/schemas/daily_report.py
import uuid
from datetime import date

from pydantic import BaseModel


class DualPanelRow(BaseModel):
    line_name: str
    daily_actual: float | None = None
    daily_forecast: float | None = None
    daily_stly: float | None = None
    daily_vs_forecast: float | None = None
    daily_vs_stly: float | None = None
    mtd_actual: float | None = None
    mtd_budget: float | None = None
    mtd_stly: float | None = None
    mtd_fcst_lock: float | None = None
    mtd_vs_budget: float | None = None
    mtd_vs_stly: float | None = None
    is_header: bool = False
    is_total: bool = False
    indent: int = 0
    unit: str = "currency"


class DailyReportSection(BaseModel):
    section_name: str
    rows: list[DualPanelRow]


class DailyReportResponse(BaseModel):
    property_id: uuid.UUID
    property_name: str
    report_date: date
    available_rooms: int
    sections: dict[str, list[DailyReportSection]]

File: frontend/src/types/daily-report.ts
export interface DualPanelRow {
  line_name: string;
  daily_actual: number | null;
  daily_forecast: number | null;
  daily_stly: number | null;
  daily_vs_forecast: number | null;
  daily_vs_stly: number | null;
  mtd_actual: number | null;
  mtd_budget: number | null;
  mtd_stly: number | null;
  mtd_fcst_lock: number | null;
  mtd_vs_budget: number | null;
  mtd_vs_stly: number | null;
  is_header: boolean;
  is_total: boolean;
  indent: number;
  unit: "currency" | "percent" | "number" | "pts";
}

export interface DailyReportSection {
  section_name: string;
  rows: DualPanelRow[];
}

export interface DailyReportResponse {
  property_id: string;
  property_name: string;
  report_date: string;
  available_rooms: number;
  sections: Record<string, DailyReportSection[]>;
}
